velox/exec/Operator.h

```diff
@@ -476,6 +478,13 @@ class Operator : public BaseRuntimeStatWriter {
   // 'identityProjections_' and 'resultProjections_'.
   RowVectorPtr fillOutput(vector_size_t size, BufferPtr mapping);
 
+  // Returns the max number of rows in the output batch.
+  virtual uint32_t outputBatchRows() const;
```

Comment:
Does this method need to be virtual?

Would you add some more details about how this method computes the result? I think it uses estimatedOutputRowSize to calculate how many rows fit in preferredOutputBatchBytes. It caps the number of rows at 10K and returns at least 1. If average row size is not know, it returns preferredOutputBatchRows.

Reply:
Thanks for the text!